apps/server/src/infra/authorization-client/authorization-client.adapter.ts

```diff
@@ -0,0 +1,58 @@
+import { Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import cookie from 'cookie';
+import { Request } from 'express';
+import { ExtractJwt, JwtFromRequestFunction } from 'passport-jwt';
+import { AuthorizationApi, AuthorizationBodyParams } from './authorization-api-client';
+import { AuthorizationErrorLoggableException, AuthorizationForbiddenLoggableException } from './error';
+
+@Injectable()
+export class AuthorizationClientAdapter {
+	constructor(private readonly authorizationApi: AuthorizationApi, @Inject(REQUEST) private request: Request) {}
+
+	public async checkPermissionsByReference(params: AuthorizationBodyParams): Promise<void> {
+		const hasPermission = await this.hasPermissionsByReference(params);
+		if (!hasPermission) {
+			throw new AuthorizationForbiddenLoggableException(params);
+		}
+	}
+
+	public async hasPermissionsByReference(params: AuthorizationBodyParams): Promise<boolean> {
+		const jwt = this.getJWT();
+
+		try {
+			const response = await this.authorizationApi.authorizationReferenceControllerAuthorizeByReference(params, {
+				headers: { authorization: `Bearer ${jwt}` },
```

Comment:
Hier wäre evtl, auch ein 
private createRequestParams(): XYZTypeVonLibaryGeklaut {
    const jwt = this.getJWT();
    const headers = { authorization: `Bearer ${jwt}` };
    
    return { headers }
}

Reply:
Fixed by implementing new function `createOptionParams`!